.ci/magician/github/set.go

```diff
@@ -149,7 +149,19 @@ func (gh *Client) MergePullRequest(owner, repo, prNumber, commitSha string) erro
 			fmt.Printf("Pull request %s is already being merged\n", prNumber)
 			return nil
 		}
-		return fmt.Errorf("failed to merge pull request: %s", err)
+		// Check if the PR is already merged (returns 405 Pull Request is not mergeable)
+		if strings.Contains(err.Error(), "Pull Request is not mergeable") {
+			fmt.Printf("Pull request %s is not mergeable; checking if it was already merged\n", prNumber)
```

Comment:
maybe we should wait 10 seconds or before we check this.

Reply:
what would that help with?

Reply:
is this a blocker?

Reply:
hmm, not a blocker. I thought it might help if there is an eventual consistency issue on the API

Reply:
I think I'd like to leave it as-is for now, in that case. If it turns out to be eventually consistent we can add a sleep later.

Reply:
Makes sense !